lib/charms/mysql/v0/backups.py

```diff
@@ -674,3 +761,130 @@ def _post_restore(self) -> Tuple[bool, str]:
+
+        if not self.charm._mysql.reconcile_binlogs_collection(
+            force_restart=True, ignore_inactive_error=True
+        ):
+            logger.error("Failed to restart binlogs collecting after S3 relation update")
+
+    def _on_s3_credentials_gone(self, event: CredentialsGoneEvent) -> None:
+        if not self.charm.unit.is_leader():
+            logger.debug("Early exit on _on_s3_credentials_gone: unit is not a leader")
+            return
+
+        self.charm.app_peer_data.update({
+            "s3-block-message": "",
+            "binlogs-collecting": "",
+        })
+        try:
+            if not self.charm._mysql.reconcile_binlogs_collection():
+                logger.error("Failed to stop binlogs collecting after S3 relation depart")
+        except Exception as e:
+            logger.error(e)
+            logger.error(
+                "Exception is occurred when trying to stop binlogs collecting after S3 relation depart. It may be a leader departure"
+            )
+
+    def get_binlogs_collector_config(self) -> Dict[str, str]:
+        """Update binlogs collector service config file.
+
+        Returns: whether this operation was successful.
+        """
+        if not self._s3_integrator_relation_exists:
+            logger.error(
+                "Cannot get binlogs collector config: s3 integrator relation does not exist"
+            )
+            return {}
+
+        logger.info("Retrieving s3 parameters from the s3-integrator relation")
+        s3_parameters, missing_parameters = self._retrieve_s3_parameters()
+        if missing_parameters:
+            logger.error(
+                f"Cannot get binlogs collector config: Missing S3 parameters: {missing_parameters}"
+            )
+            return {}
+
+        bucket_url = (
+            f"{s3_parameters['bucket']}/{s3_parameters['path']}binlogs"
+            if not s3_parameters["path"] or s3_parameters["path"][-1] == "/"
+            else f"{s3_parameters['bucket']}/{s3_parameters['path']}/binlogs"
+        )
+
+        return {
+            "ENDPOINT": _construct_endpoint(s3_parameters),
+            "HOSTS": ",".join(self.charm._mysql.get_cluster_members()),
+            "USER": SERVER_CONFIG_USERNAME,
+            "PASS": self.charm.get_secret("app", SERVER_CONFIG_PASSWORD_KEY),
+            "STORAGE_TYPE": "s3",
+            "ACCESS_KEY_ID": s3_parameters["access-key"],
+            "SECRET_ACCESS_KEY": s3_parameters["secret-key"],
+            "S3_BUCKET_URL": bucket_url,
+            "DEFAULT_REGION": s3_parameters["region"],
+        }
+
+    def _is_mysql_timestamp(self, timestamp: str) -> bool:
+        # Format is the same as in the mysql-pitr-helper project.
+        if not re.match(
+            r"^\d{4}-\d{2}-\d{2} \d{2}:\d{2}:\d{2}$",
+            timestamp,
+        ):
+            return False
+        try:
+            self._parse_mysql_timestamp(timestamp)
+            return True
+        except ValueError:
+            return False
```

Comment:
What's wrong with just relying on the `try-except` block? Why do we need to regex-match? 🤔

Reply:
good catch. @Zvirovyi  see any issues with :

```suggestion
        try:
            datetime.datetime.strptime(timestamp, "%Y-%m-%d %H:%M:%S")
            return True
        except ValueError:
            return False
```

Reply:
With the suggested approach, it will accept values such as `2025-2-02 1:10:10` in Python, while `time.Parse` in [mysql-pitr-helper](https://github.com/canonical/mysql-pitr-helper/blob/ed858df5c145b003c9d24223d44b6ea9c7d67888/recoverer/recoverer.go#L194) (Go) will return an error ([example](https://go.dev/play/p/uvtfFi1CLrR)). My advice is to keep regex to enforce a strict format across different projects/languages regardless of their own time-parsing implementation.

On the other hand, we can loosen the check on the mysql-pitr-helper side, but it will require testing to ensure full compatibility and I don't see it viable.

Reply:
Thanks for clarifying.

I would probably add a test failing if the string date does not follow the plugin expected format, or a comment stating why the regex is there (i.e. required by the Go plugin 2-3 abstraction layers deeper). Otherwise, I am afraid this snippet may be mistakenly removed in the future.

Reply:
I'll add a comment